homeassistant/components/mobile_app/websocket_api.py

```diff
@@ -0,0 +1,85 @@
+"""Websocket API for mobile_app."""
+from aiohttp.http_websocket import WSMessage
+
+from homeassistant.components.cloud import (async_delete_cloudhook,
+                                            async_is_logged_in)
+from homeassistant.components.websocket_api import (ActiveConnection,
+                                                    async_register_command,
+                                                    async_response,
+                                                    error_message,
+                                                    result_message,
+                                                    websocket_command,
+                                                    ws_require_user)
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (ATTR_DELETED_IDS, ATTR_REGISTRATIONS, ATTR_STORE,
+                    CONF_USER_ID, CONF_WEBHOOK_ID, DOMAIN,
+                    SCHEMA_WS_DELETE_REGISTRATION, SCHEMA_WS_GET_REGISTRATION)
+
+from .helpers import safe_device, savable_state
+
+
+def register_websocket_handlers(hass: HomeAssistantType) -> bool:
+    """Register the websocket handlers."""
+    async_register_command(hass, websocket_get_registration)
+
+    async_register_command(hass, websocket_delete_registration)
+
+    return True
+
+
+@ws_require_user()
+@async_response
+@websocket_command(SCHEMA_WS_GET_REGISTRATION)
+async def websocket_get_registration(
+        hass: HomeAssistantType, connection: ActiveConnection,
+        msg: WSMessage) -> None:
+    """Return the registration for the given webhook_id."""
+    user = connection.user
+
+    webhook_id = msg[CONF_WEBHOOK_ID]
+
+    device = hass.data[DOMAIN][ATTR_REGISTRATIONS][webhook_id]
```

Comment:
What if webhook_id doesn't exist ? 

Also, I would expect for an editor in the UI that we need:

 - list all registrations for a user (that way an admin can use that to request for everyone) (only admin can pick a user that is not itself)
 - delete registration

Reply:
Delete registration is already there.